### PR TITLE
ci: move Azure to GHA (Android)

### PR DESCRIPTION
**Pull request by KazuCocoa**
Let's run one first test on GHA from Azure.

This includes only  `test/functional/android/device_time_tests.py test/functional/android/search_context/find_by_*.py`

### CODE REVIEW

File: .github/workflows/functional-test.yml [hunk 1]
```diff
@@ -27,6 +27,11 @@ jobs:
 
     runs-on: macos-14
 
+    env:
+      XCODE_VERSION: 15.3
```
Comment (mykola-mokhnach) [inline .github/workflows/functional-test.yml:31]: maybe add a note that in order to change xcode version one must also check for compatible macOS version in the compat matrix

File: .github/workflows/functional-test.yml [hunk 2]
```diff
@@ -38,14 +43,14 @@ jobs:
...
-        xcode-version: 15.3
+        xcode-version: ${{ env.XCODE_VERSION }}
     - run: defaults write com.apple.iphonesimulator PasteboardAutomaticSync -bool false
 
     - uses: futureware-tech/simulator-action@v3
       with:
         # https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md
-        model: 'iPhone 15 Plus'
-        os_version: '17.4'
+        model: ${{ env.IPHONE_MODEL }}
```
Comment (mykola-mokhnach) [inline .github/workflows/functional-test.yml:52]: should these values be quoted?

File: .github/workflows/functional-test.yml [hunk 3]
```diff
@@ -77,3 +82,101 @@ jobs:
...
+        target: google_apis
+        force-avd-creation: false
+        emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+        disable-animations: false
+        script: echo "Generated AVD snapshot for caching."
+
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.9
```
Comment (mykola-mokhnach) [inline .github/workflows/functional-test.yml:152]: should it rather be latest python?

File: .github/workflows/functional-test.yml [hunk 4]
```diff
@@ -77,3 +82,101 @@ jobs:
...
+        node-version: 'lts/*'
+
+    # Start Appium
+    - run: npm install -g appium
+    - run: |
+        appium driver install uiautomator2
+        appium driver install espresso
+        appium plugin install images
+        appium plugin install execute-driver
+        nohup appium --use-plugins=images,execute-driver --relaxed-security --log-timestamp --log-no-colors > appium.log &
```
Comment (mykola-mokhnach) [inline .github/workflows/functional-test.yml:121]: we should also redirect the stderr output to stdout

File: .github/workflows/functional-test.yml (hunk 2 above)
Comment (KazuCocoa) [reply .github/workflows/functional-test.yml:52]: Maybe either is fine? This works right now. I'm good either way.

Comment (mykola-mokhnach) [reply .github/workflows/functional-test.yml:54]: fine with me if it works this way

File: .github/workflows/functional-test.yml (hunk 3 above)
Comment (mykola-mokhnach) [reply .github/workflows/functional-test.yml:154]: np, just leave a TODO there for later so we don't forget about it.

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**